### PR TITLE
Add support for synapse workspace and synapse big data pool

### PR DESCRIPTION
**Pull request by emilychu9318**
**What this PR does / why we need it**:
This PR enables the Synapse workspace and workspace bigDataPools resource in code generation and adds a test that exercises creating, updating and deleting it.

**How does this PR make you feel**:

**If applicable**:
- [:white_check_mark:] this PR contains tests

### CODE REVIEW

File: v2/samples/synapse/v1beta/v1beta20210601_synapsebigdatapool.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apiVersion: synapse.azure.com/v1beta20210601
```
Comment (super-harsh) [inline v2/samples/synapse/v1beta/v1beta20210601_synapsebigdatapool.yaml:1]: Nice! you'd also need to run the tests on these samples to generate a recording. You can follow the [documentation](https://azure.github.io/azure-service-operator/contributing/add-a-new-code-generated-resource/#run-test-for-added-sample-and-commit-the-recording) to do so.

**Review by super-harsh — COMMENTED** [review]
Thanks for your contribution! 

It looks pretty good. However, I've mentioned a few minor things below which would require some changes.

File: v2/api/datalakestore/customizations/structure.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+github.com/Azure/azure-service-operator/v2/api/datalakestore/customizations
```
[the remark this answers is not in the document]
Comment (emilychu9318) [reply v2/api/datalakestore/customizations/structure.txt:1]: I think v2/api/dalalakestore folder is not auto updated after I removed datalakestore from azure-arm.yaml, I will delete this folder.

**Review by matthchr — COMMENTED** [review]
Thanks for this contribution! As @super-harsh said it looks great, just had a few small comments.

File: v2/internal/controllers/crd_synapes_test.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	// Create a storage account to use as workspace datalake storage
+	kind := storage.StorageAccount_Kind_Spec_StorageV2
+	sku := storage.SkuName_Standard_LRS
+	hnsEnabled := true
+	acct := &storage.StorageAccount{
+		ObjectMeta: tc.MakeObjectMetaWithName(tc.NoSpaceNamer.GenerateName("adlstore")),
+		Spec: storage.StorageAccount_Spec{
+			Location:     &lc,
+			Owner:        testcommon.AsOwner(rg),
+			Kind:         &kind,
```
[the remark this answers is not in the document]
Comment (emilychu9318) [reply v2/internal/controllers/crd_synapes_test.go:34]: I couldn't find to.Ptr [here](https://github.com/Azure/go-autorest/blob/main/autorest/to/convert.go)

[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/internal/controllers/crd_synapes_test.go:34]: We used to use the AutoRest versions, but that was removed in #2817 (as we weren't using the rest of it) - now we have our own helper package. 

Import `github.com/Azure/azure-service-operator/v2/internal/util/to`.
```

File: v2/internal/controllers/crd_synapes_test.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	// Create a workspace for bib data pool
+	identityType := synapse.ManagedIdentity_Type_SystemAssigned
+	ws := &synapse.Workspace{
+		ObjectMeta: tc.MakeObjectMeta("workspace"),
+		Spec: synapse.Workspace_Spec{
+			Identity: &synapse.ManagedIdentity{
+				Type: &identityType,
+			},
+			DefaultDataLakeStorage: &synapse.DataLakeStorageAccountDetails{
+				AccountUrl: acct.Status.PrimaryEndpoints.Dfs,
```
Comment (matthchr) [inline v2/internal/controllers/crd_synapes_test.go:51]: The fact that you're linking these two things together by creating one resource and then including the URL in the second resource is an indication that this `AccountUrl` parameter should also have an `$importConfigMapMode: optional` set on it in `azure-arm.yaml`. This will allow users to import the URL from a configmap. See [here](https://github.com/Azure/azure-service-operator/blob/main/v2/azure-arm.yaml#L1487) for example.

The actual entry for this URL should be something like:
```
      DataLakeStorageAccountDetails:
        AccountUrl:
          $importConfigMapMode: optional
```

If you have any issues with that, reach out over Slack and we can help.

The other side of this is that the storage account URL needs to be actually exportable to a ConfigMap. Right now it's exportable to a Secret but not to a ConfigMap. That's partially because when we added storage accounts we didn't support exporting to ConfigMaps yet. I think the right thing to do is to _also_ allow exporting the storage accounts endpoints to a configmap. I've sent a PR to do that here: #2855

TL;DR: Can you do three things here:
1. Create the resources "all at once" in this test as I mention in my comment here: https://github.com/Azure/azure-service-operator/pull/2851/files#r1161037489
2. Label the `AccountUrl` as importable from a configMap as above.
3. Once my PR #2855 is merged, rebase this onto it and export the storage account endpoint to a configmap, and then import that configmap value into accountURL here.

**Comment by theunrepentantgeek** [comment]
@emilychu9318, if you're willing to reopen this PR, we'd like to work with you to get this landed in ASO.

**Comment by emilychu9318** [comment]
Sorry this pr get automatically closed after I did a resync of my branch to main, please refer this new pr: https://github.com/Azure/azure-service-operator/pull/2860
